picard/ui/metadatabox.py

```diff
@@ -309,16 +309,29 @@ def contextMenuEvent(self, event):
             if len(tags) == 1 or removals or useorigs:
                 menu.addSeparator()
             menu.addAction(self.add_tag_action)
+            add_to_preserved_tags_action = QtGui.QAction(_(u"Add Selected Tags to 'Preserve Tags' List"), self.parent)
+            add_to_preserved_tags_action.triggered.connect(self.add_to_preserved_tags)
+            menu.addAction(add_to_preserved_tags_action)
             menu.addSeparator()
         menu.addAction(self.changes_first_action)
         menu.exec_(event.globalPos())
         event.accept()
 
+    def add_to_preserved_tags(self):
+        # Get all selected tags including '~length'
+        tags = self.selected_tags()
+        preserved_tags = [tag for tag in
+                          map(lambda x: x.strip(), config.setting['preserved_tags'].split(','))
+                          if tag != ""]
+        preserved_tags.extend(tags)
+        config.setting['preserved_tags'] = ", ".join(filter(lambda x: x in TAG_NAMES.keys(),
```

Comment:
Minor nit: this filter is a bit too complicated, you could just write (I did not actually test this ;-) ):

```python
[...].join(tag for tag in set(preserved_tags) if tag in TAG_NAMES.keys())
```